### DISCUSSION TITLE
Matrix Issue

### DISCUSSION THREAD

**Discussion by MikeTX01** [MLB-LED-Scoreboard/mlb-led-scoreboard#531] [General]
So, I'm using a 64x32 P3 Waveshare matrix.  Im using a Raspberry Pi 2 W and the Adafruit RGB Matrix + RTC Hat.  I have changed the appropriate file in the coordinate folder and i have tried the command lines i have seen in other questions.  Nothing seems to work, im constantly having everything display as if its 32x32 (giving me double screen) .  Also maybe i didnt edit the main file correctly because its streaming multiple other games while my team is playing.

**Comment by ty-porter** [comment]
You will need to use `--led-cols=64` in your startup command. The software does not automatically detect your matrix size.

**Reply by MikeTX01** [reply]
I get this error when i tried that:

~/mlb-led-scoreboard $ sudo ./main.py --led-gpio-mapping="adafruit-hat" --led-cols=64
Suggestion: to slightly improve display update, add
        isolcpus=3
at the end of /boot/cmdline.txt and reboot (see README.md)
ERROR (03:55:21): Invalid matrix dimensions provided. See top of README for supported dimensions.
If you would like to see new dimensions supported, please file an issue on GitHub!
ERROR (03:55:21): Untrapped error in main!
Traceback (most recent call last):
  File "/home/family/mlb-led-scoreboard/./main.py", line 169, in <module>
    main(matrix, config)
  File "/home/family/mlb-led-scoreboard/./main.py", line 41, in main
    config = Config(config_base, matrix.width, matrix.height)
  File "/home/family/mlb-led-scoreboard/data/config/__init__.py", line 85, in __init__
    json = self.__get_layout(width, height)
  File "/home/family/mlb-led-scoreboard/data/config/__init__.py", line 244, in __get_layout
    sys.exit(1)
SystemExit: 1

**Reply by ty-porter** [reply]
Did you delete or rename `w64h32.json.example` when creating a custom config? If so, **do not** do that, copy the file to `w64h32.json` instead.

**Reply by MikeTX01** [reply]
i did just rename that file to w64h32.json  so should i rename it back to *.example and then copy and rename?   im doing everything over putty and not any FTP.

**Reply by ty-porter** [reply]
Yes, those are reference files. They must exist for the scoreboard to support the size.

**Reply by MikeTX01** [reply]
so as in copy make a duplicate and rename it to the w64h32.json, and leave it in the same folder?

**Reply by ty-porter** [reply]
You **must** have `w64h32.json.example` in the `coordinates` directory. If you want a custom layout, you can override it by creating `w64h32.json`.

If you do not care about a custom layout, `w64h32.json` is not needed.

**Reply by MikeTX01** [reply]
I just didn't know it that was a way to have it automatically go to the proper size when you started without having to add --led-cols=64.

also, every time I close out telnet the matrix/scoreboard has issues and doesn't work anymore.  is there a way to fix that as well?

**Reply by ty-porter** [reply]
You are going to have to use the `--led-cols` flag, again there is no automatic detection of what size matrix you use.

To have the board stay on, you will either need to leave the terminal open with the script running or run it as a daemon:
https://github.com/MLB-LED-Scoreboard/mlb-led-scoreboard/wiki/Creating-a-systemd-startup-script

**Reply by MikeTX01** [reply]
i have followed the instructions and it will not auto start at all.

**Reply by MikeTX01** [reply]
Auto start just wont working i have tried both 7 and 8 options.

**Reply by ty-porter** [reply]
Did you change the startup arguments in the service to match what your matrix needs?

Have you checked the service is running?

```
$  sudo systemctl status mlb-led-scoreboard
```

Have you done any troubleshooting at all via the logs?

```
$ sudo journalctl -u mlb-led-scoreboard --since "1 day ago"
```
